examples/features/nse-composition/README.md

```diff
@@ -0,0 +1,318 @@
+Create test namespace:
+```bash
+NAMESPACE=($(kubectl create -f ../namespace.yaml)[0])
+NAMESPACE=${NAMESPACE:10}
+```
+
+Register namespace in `spire` server:
+```bash
+kubectl exec -n spire spire-server-0 -- \
+/opt/spire/bin/spire-server entry create \
+-spiffeID spiffe://example.org/ns/${NAMESPACE}/sa/default \
+-parentID spiffe://example.org/ns/spire/sa/spire-agent \
+-selector k8s:ns:${NAMESPACE} \
+-selector k8s:sa:default
+```
+
+Select node to deploy NSC and NSE:
+```bash
+NODE=($(kubectl get nodes -o go-template='{{range .items}}{{ if not .spec.taints  }}{{index .metadata.labels "kubernetes.io/hostname"}} {{end}}{{end}}')[0])
+```
+
+Create Passthrough and firewall NSE configurations:
+```bash
+PASS_COUNT=4
+PASS_CFG=""
+NAME=""
+LABELS=""
+VOLUME_PATCH=""
+MATCHES=""
+
+for ((i = 1; i <= PASS_COUNT; i++))
+do
+  f="passthrough-${i}"
+  NAME="nse-passthrough-${i}"
+  LABELS="app:passthrough-${i}"
+  if ((i == PASS_COUNT))
+  then
+    f="nse-firewall"
+    NAME="acl-filter"
+    LABELS="app:firewall"
+    VOLUME_PATCH="- config-patch.yaml"
+    MATCHES="
+    - source_selector:
+        app: firewall${MATCHES}
+      routes:
+        - destination_selector:
+            app: gateway
+    - routes:
+        - destination_selector:
+            app: firewall"
+  else
+    MATCHES="${MATCHES}
+      routes:
+        - destination_selector:
+            app: passthrough-${i}
+    - source_selector:
+        app: passthrough-${i}"
+  fi
+  if [ -d "${f}" ]
+  then
+    rm -r "${f}"
+  fi
+  mkdir "${f}"
+  cat > "${f}"/kustomization.yaml <<EOF
+---
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+bases:
+- ../../../../apps/nse-firewall-vpp
+patchesStrategicMerge:
+- patch-nse-firewall-vpp.yaml
+${VOLUME_PATCH}
+patches:
+- target:
+    kind: Deployment
+    name: nse-firewall-vpp
+  patch: |-  
+    - op: replace  
+      path: /metadata/name
+      value: ${NAME}
+EOF
+if((i == PASS_COUNT))
+then
+cat > "${f}"/config-patch.yaml <<EOF
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nse-firewall-vpp
+spec:
+  template:    
+    spec:
+      containers:
+        - name: nse  
+          volumeMounts:
+            - mountPath: /etc/vppagent-firewall/config.yaml
+              subPath: config.yaml
+              name: vppagent-firewall-config-volume
+      volumes:
+        - name: vppagent-firewall-config-volume
+          configMap:
+            name: vppagent-firewall-config-file
+EOF
+fi
+  cat > "${f}"/patch-nse-firewall-vpp.yaml <<EOF
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nse-firewall-vpp
+spec:
+  template:
+    spec:
+      containers:
+        - name: nse
+          env:
+            - name: NSM_SERVICE_NAME
+              value: "nse-composition"
+            - name: NSM_LABELS
+              value: ${LABELS}
+      nodeSelector:
+        kubernetes.io/hostname: ${NODE}
+EOF
+  PASS_CFG="${PASS_CFG}- ./${f}"
+  if (( i < PASS_COUNT ))
+  then
+    PASS_CFG+=$'\n'
+  fi
+done
```

Comment:
This bash command looks complicated.

Question: Can we define passthrough nses staticaly?

Reply:
yes, we can
simplified script, added static files